jcl/src/openj9.sharedclasses/share/classes/com/ibm/oti/shared/SharedClassPermission.java

```diff
@@ -27,75 +20,82 @@
+ *
+ * <ul>
+ * <li>"read" allows a ClassLoader to load classes from the shared cache</li>
+ * <li>"write" allows a ClassLoader to add classes to the shared cache</li>
+ * </ul>
  */
 public class SharedClassPermission extends BasicPermission {
 
 	private static final long serialVersionUID = -3867544018716468265L;
 
-	transient private boolean read, write;
-	
+	private transient boolean read;
+	private transient boolean write;
+
 	/**
 	 * Constructs a new instance of this class.
-	 * <p>
-	 * @param		loader java.lang.ClassLoader.
-	 *					The ClassLoader requiring the permission.
-	 * @param		actions java.lang.String.
-	 *					The actions which are applicable to it.
+	 *
+	 * @param loader ClassLoader the ClassLoader requiring the permission
+	 * @param actions String the actions which are applicable to it
 	 */
 	public SharedClassPermission(ClassLoader loader, String actions) {
 		this(loader.getClass().getName(), actions);
 	}
 
 	/**
 	 * Constructs a new instance of this class.
-	 * <p>
-	 * @param		classLoaderClassName java.lang.String.
-	 *					The className of the ClassLoader requiring the permission.
-	 * @param		actions java.lang.String.
-	 *					The actions which are applicable to it.
+	 *
+	 * @param classLoaderClassName String the name of the ClassLoader class requiring the permission
+	 * @param actions String the actions which are applicable to it
 	 */
 	public SharedClassPermission(String classLoaderClassName, String actions) {
 		super(classLoaderClassName);
 		decodeActions(actions);
 	}
 
 	private void decodeActions(String actions) {
-		StringTokenizer tokenizer =
-			new StringTokenizer(actions.toLowerCase(), " \t\n\r,"); //$NON-NLS-1$
+		StringTokenizer tokenizer = new StringTokenizer(actions.toLowerCase(), " \t\n\r,"); //$NON-NLS-1$
 		while (tokenizer.hasMoreTokens()) {
 			String token = tokenizer.nextToken();
-			if (token.equals("read")) //$NON-NLS-1$
+			if (token.equals("read")) { //$NON-NLS-1$
 				read = true;
-			else if (token.equals("write")) //$NON-NLS-1$
+			} else if (token.equals("write")) { //$NON-NLS-1$
 				write = true;
-			else
+			} else {
 				throw new IllegalArgumentException();
+			}
+		}
+		if (!read && !write) {
+			throw new IllegalArgumentException();
 		}
-		if (!read && !write) throw new IllegalArgumentException();
 	}
 
 	/**
 	 * Compares the argument to the receiver, and answers <code>true</code>
 	 * if they represent the <em>same</em> object using a class
-	 * specific comparison. In this case, the receiver must be
-	 * for the same property as the argument, and must have the
-	 * same actions.
-	 * <p>
-	 * @param		o		The object to compare with this object
-	 * @return		<code>true</code>
-	 *					If the object is the same as this object
-	 *				<code>false</code>
-	 *					If it is different from this object
-	 * @see			#hashCode
+	 * specific comparison.
+	 *
+	 * <p>In this case, the receiver must be for the same property as the argument,
```

Comment:
Is there a reason to add `<p>` to separate the 2 sentences ? I think this sentence is explaining the `true` case in the above sentence.

Reply:
Javadoc is structured so that the first sentence provides a summary. Only the summary appears in the table of methods. The summary and the following paragraphs are included in the separate, detailed description. They are already separated: the explicit `<p>` improves the appearance in the 'Javadoc' view within eclipse.

Reply:
OK, thanks.